client/rest-high-level/src/main/java/org/elasticsearch/client/indices/AnalyzeResponse.java

```diff
@@ -0,0 +1,259 @@
+            if (o == null || getClass() != o.getClass()) return false;
+            AnalyzeResponse.AnalyzeToken that = (AnalyzeResponse.AnalyzeToken) o;
+            return startOffset == that.startOffset &&
+                endOffset == that.endOffset &&
+                position == that.position &&
+                positionLength == that.positionLength &&
+                Objects.equals(term, that.term) &&
+                Objects.equals(attributes, that.attributes) &&
+                Objects.equals(type, that.type);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(term, startOffset, endOffset, position, positionLength, attributes, type);
+        }
+
+        AnalyzeToken(String term, int position, int startOffset, int endOffset, int positionLength,
+                     String type, Map<String, Object> attributes) {
+            this.term = term;
+            this.position = position;
+            this.startOffset = startOffset;
+            this.endOffset = endOffset;
+            this.positionLength = positionLength;
+            this.type = type;
+            this.attributes = attributes;
+        }
+
+        public String getTerm() {
+            return this.term;
+        }
+
+        public int getStartOffset() {
+            return this.startOffset;
+        }
+
+        public int getEndOffset() {
+            return this.endOffset;
+        }
+
+        public int getPosition() {
+            return this.position;
+        }
+
+        public int getPositionLength() {
+            return this.positionLength;
+        }
+
+        public String getType() {
+            return this.type;
+        }
+
+        public Map<String, Object> getAttributes() {
+            return this.attributes;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(AnalyzeResponse.Fields.TOKEN, term);
+            builder.field(AnalyzeResponse.Fields.START_OFFSET, startOffset);
+            builder.field(AnalyzeResponse.Fields.END_OFFSET, endOffset);
+            builder.field(AnalyzeResponse.Fields.TYPE, type);
+            builder.field(AnalyzeResponse.Fields.POSITION, position);
+            if (positionLength > 1) {
+                builder.field(AnalyzeResponse.Fields.POSITION_LENGTH, positionLength);
+            }
+            if (attributes != null && !attributes.isEmpty()) {
+                Map<String, Object> sortedAttributes = new TreeMap<>(attributes);
+                for (Map.Entry<String, Object> entity : sortedAttributes.entrySet()) {
+                    builder.field(entity.getKey(), entity.getValue());
+                }
+            }
+            builder.endObject();
+            return builder;
+        }
+
+        public static AnalyzeResponse.AnalyzeToken fromXContent(XContentParser parser) throws IOException {
```

Comment:
We should definitely use a `ConstructingObjectParser` here if possible, instead of this hand rolled `fromXContent`

Reply:
The tricky thing here is that AnalyzeToken has a small number of defined fields, but then scoops up all other fields defined at the same level into an `attributes` map; our object parsers don't currently have a way of doing that.

Reply:
ugh, fun. Lets make a mental note to fix this in the next version of the API :)